src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/WorkspaceSemanticTokensRefreshPublisher.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.Razor;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    internal abstract class WorkspaceSemanticTokensRefreshPublisher : IDisposable
```

Comment:
The abstract class being disposable is a detail that the service collection will take care of. I wouldn't bake `IDisposable` into the `abstract class` itself